### ISSUE TITLE
Surprising empty output in skimage.feature.hog()

### ISSUE THREAD

**Issue by crisluengo** [scikit-image/scikit-image#7064]
### Description:

If the image is too small given the chosen `pixels_per_cell` and `cells_per_block`, an empty array is produced as output. [This is a surprising output.](https://stackoverflow.com/q/76681370/7328782)

I suggest you produce a descriptive error message instead.

Alternatively, make this behavior clear in the documentation. The documentation could, by the way, show how to compute the expected output size of the array, as this is not obvious from the outside.

### Way to reproduce:

```python
import numpy as np
import skimage

img = np.zeros((16, 193))
print(skimage.feature.hog(img))
```

### Version information:

I'm using scikit-image version 0.21.0.

[The remainder of the information is irrelevant, I can see [in the source code where this is happening](https://github.com/scikit-image/scikit-image/blob/76dd2dab856f95e5264f50e376f9f3a81a261287/skimage/feature/_hog.py#L277).]

**Comment by lagru** [comment]
Thanks for the report and complete diagnosis! I agree, that we should make this case less surprising to users and raise an error. Do you want to do the "honors"? No pressure or expectation of course! :D

**Comment by crisluengo** [comment]
Hi @lagru 
I'm happy to make these changes, but I'm not sure when I'll have the time do to it.

**Comment by lagru** [comment]
No worries. I took the liberty to mark this as a good first issue in case someone else wants to take a shot. I'd be happy with a descriptive error message or maybe a descriptive warning. And in any case it never hurts to describe the behavior in more detail in the docstring. :+1:

**Comment by crisluengo** [comment]
This seems fixed in #7153.
